iCount/analysis/rnamaps.py

```diff
@@ -3,690 +3,233 @@
-
-        * Change the reference sequence? This can be complex... Make a helper
-          tool for this?
-        * Provide additional data in function - which exceptions /abnormalities
-          to ignore?
-
-    * Gaussian smooting of RNAmaps?
-
-    * split read - it can differ also on "first-stop", not juts second-start
-        * we could have some sort of quality check when observing the variance
-          of first-stop-s. THe major question is: "Do all reads for certain
-          xlink event indicate the same behaviour?"
-
+Perform RNA-maps analysis.
 """
+import csv
 import logging
+import os
 
-import pybedtools
-import matplotlib
-matplotlib.use('Agg')
-import matplotlib.pyplot as plt  # pylint: disable=wrong-import-position
+from pybedtools import BedTool, create_interval_from_list
 
 import iCount  # pylint: disable=wrong-import-position
-from iCount.files import _f2s  # pylint: disable=wrong-import-position
 
 LOGGER = logging.getLogger(__name__)
 
-EXON_TYPES = ['CDS', 'ncRNA', 'UTR3', 'UTR5']
-RNA_WINDOW_SIZE = 2000  # TODO: rethink how this constant would affect lengths in RNAmap generation
-
-
-def _add_entry(start_type, stop_type, distance, score, strand, data, metrics, explict=False):
-    """Add RNA-map entry in ``data``."""
-    if strand == '+':
-        rna_map_type = start_type + '-' + stop_type
-    else:
-        rna_map_type = stop_type + '-' + start_type
-        distance = -distance
-
-    data.setdefault(rna_map_type, {}).setdefault(distance, [0, 0])[0] += score
-    if explict:
-        data.setdefault(rna_map_type, {}).setdefault(distance, [0, 0])[1] += score
-
-    # Increment also mRNA / pre-mRNa counters:
-    if 'intron' in rna_map_type or 'intergenic' in rna_map_type:
-        metrics.origin_premrna += score
-    elif start_type in EXON_TYPES and stop_type in EXON_TYPES:
-        metrics.origin_mrna += score
-    else:
-        metrics.origin_ambiguous += score
-
-
-def _process_read_group(xlink, chrom, strand, read, data, segmentation, metrics,
-                        implicit_handling='closest'):
-    """
-    Process each read group.
-
-    Each group is represented by read. It has a fixed xlink, start, second-start
-    and end position. This function finds all landmarks that this read is
-    overlapping. It also determines for each of them:
+MAP_TYPES = {
+    'exon-intron': {
+        'upstream-type': ['CDS'],
+        'upstream-size-limit': 50,
+        'upstream-plot-width': 50,
+        'downstream-type': ['intron'],
+        'downstream-size-limit': 150,
+        'downstream-plot-width': 100,
+    },
+    'intron-exon': {
+        'upstream-type': ['intron'],
+        'upstream-size-limit': 150,
+        'upstream-plot-width': 100,
+        'downstream-type': ['CDS'],
+        'downstream-size-limit': 50,
+        'downstream-plot-width': 50,
+    },
+    'gene-start': {
+        'upstream-type': ['intergenic'],
+        'upstream-plot-width': 200,
```

Comment:
Hi Jure, shouldn't you also set the 'upstream-size-limit': 200 here? This applies to lines up to 69, where each plot width should be coupled with size limit. In other words, we should ensure that each plot only collects data from regions that are at least as wide as the width of the plot.

Reply:
Sure, no problem. When we were discussing how to implement this, i understood that size limits apply only for `exon-intron` and `intron-exon` types. It is even better the way you suggest, since this removes the distinction between `plot-width` and `size-limit` parameters.

Reply:
Good - just one point - in case of introns, plot-width and size-limit are of different size (100 and 150), and that is correct. In all other regions, plot-width and size-limit can be of same size.